### DISCUSSION TITLE
Improve usability of togglable icon buttons

### DISCUSSION THREAD

**Discussion by miv391** [godotengine/godot-proposals#4852] [Editor]
Currently it is pretty much impossible to see the difference between most selected and non-selected icons. For example icon buttons used to select Path2D's edit modes:

![kuva](https://user-images.githubusercontent.com/49998025/178271668-4a2ae9d5-8e96-4cbc-8732-389569154230.png)

In the first group the first button is selected, in the second group the second button is selected. It is faster and less error prone just to click the button instead of trying to see if the desired mode is already selected. This is of course very bad usability.

**Proposal**: Change selected icons so that the selection is clearly seen. Maybe a background color with high contrast to the current background color. Or bright yellow top border or full border. If the Godot devs want to keep the current look & feel regardless of the usability, then make this feature as an accessibility feature which can be enabled if wanted. In that case maximal visibility should be the goal, e.g. the yellow border could be good.

Yellow top border and yellow full border:

![kuva](https://user-images.githubusercontent.com/49998025/178272414-727d9eb7-bdf0-4d19-a705-3281aaed616e.png)

**Comment by Calinou** [comment]
In an earlier revision of the [improved editor theme](https://github.com/godotengine/godot/pull/45607), I added a colored background to all pressed buttons (including flat buttons). I removed it for some reason – I don't remember why, but you may find the reason by reading the comments and collapsed sections in that PR.

> If the Godot devs want to keep the current look & feel regardless of the usability, then make this feature as an accessibility feature which can be enabled if wanted.

This should *not* be an option, as good usability should be always enabled :slightly_smiling_face:

**Reply by miv391** [reply]
I read that PR but couldn't find the exact reason why buttons were made flat. I think that the non-flat button looked pretty good and their usability is of course much better. If flat look is preferred, I also like my own "yellow top border" suggestion.

While reading that PR I noticed that two slightly different colors are used as a selection/highlight color. I wonder if that is done on purpose?

![kuva](https://user-images.githubusercontent.com/49998025/179035862-9a08c8b3-6c48-41e3-bc92-853d5877d5a6.png)

**Reply by Calinou** [reply]
> While reading that PR I noticed that two slightly different colors are used as a selection/highlight color. I wonder if that is done on purpose?

I believe this was done on purpose because some people found the tab highlight too distracting. I didn't want to remove the tab highlight entirely though, as it does serve a purpose (making it easier to see which tab is currently selected).
